modules/gapi/src/backends/cpu/gcpubackend.cpp

```diff
@@ -176,17 +227,28 @@ void cv::gimpl::GCPUExecutable::run(std::vector<InObj>  &&input_objs,
 
         // Obtain our real execution unit
         // TODO: Should kernels be copyable?
+
         GCPUKernel k = gcm.metadata(op_info.nh).get<Unit>().k;
 
         // Initialize kernel's execution context:
         // - Input parameters
         GCPUContext context;
-        context.m_args.reserve(op.args.size());
-
-        using namespace std::placeholders;
-        ade::util::transform(op.args,
-                          std::back_inserter(context.m_args),
-                          std::bind(&GCPUExecutable::packArg, this, _1));
+        if (k.m_hasState)
+        {
+            if (m_needKernelsSetup)
+            {
+                setupKernelState(op_info.nh, context);
+            }
+            else
+            {
+                fillContextArgs(op.args, context.m_args);
+                context.m_statePtr = &m_nodesToStates[op_info.nh];
+            }
+        }
+        else
+        {
+            fillContextArgs(op.args, context.m_args);
+        }
```

Comment:
This is somewhat complicated.

Instead, you could check your flag once (before the loop), and if you need to reset the state, then walk through the _affected_ nodes only and call the callback.

Please bring that old 6 lines of code back here.

Reply:
I walked away from implementation with handling of only affected nodes because thought in the way that runtime information may be needed for setup. If last was the right so we need to retrieve the newest runtime data before the setup call.

Reply:
Implemented handling of only affected nodes and moved this logic upper in the "run" code.